### PR TITLE
Anyhow migration

### PR DESCRIPTION
**Pull request by rbtcollins**
error_chain is giving us more and more headaches; time to let it pass beyond the vale.

### CODE REVIEW

File: src/cli/common.rs [hunk 1]
```diff
@@ -3,9 +3,11 @@
 use crate::errors::*;
 use crate::self_update;
 use crate::term2;
+use anyhow;
```
Comment (lzutao) [inline src/cli/common.rs]: We shouldn't need this use statement.

File: src/errors.rs [hunk 2]
```diff
@@ -370,7 +341,168 @@ error_chain! {
...
+#[derive(ThisError, Debug)]
+pub enum RustupError {
+    #[error("invalid toolchain name: '{0}'")]
+    InvalidToolchainName(String),
+    #[error("Unable to proceed. Could not locate working directory.")]
+    LocatingWorkingDir {
+        #[from]
+        source: io::Error,
+    },
+    #[error("failed to set permissions for '{}'", .p.display())]
```
Comment (dtolnay) [inline src/errors.rs:342]:
```suggestion
    #[error("failed to set permissions for '{p}'")]
```

We'll display the path automatically.

**Comment by kinnison** [comment]
@rbtcollins Are you likely to be able to do any more on this any time soon?  it would be nice to get it merged before we end up pushing changes which make it harder to rebase.

**Comment by kinnison** [comment]
I have a feeling that migrating many of these patches will now be a pain with the rebase needed, however if we can get it to a point that it's working then I'd love to get more of this migration done.

**Pull request closed without merging**